sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,119 @@
+import uuid
+import warnings
+import random
+import os
 
+STORAGE_CONN_STR = [os.environ.get("AZURE_STORAGE_CONN_STR", "Azure Storage Connection String")]
 
-def test_constructor():
-    client = TableCheckpointStore()
-    assert client is not None
-    
+def get_live_storage_table_client(storage_connection_str):
+    try:
+        table_name = chr(random.randint(ord('a'), ord('z'))) +(uuid.uuid4().hex)
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.create_table_if_not_exists(table_name)
+        return storage_connection_str, table_name
+    except:
+        pytest.skip("Storage table client can't be created")
+
+def remove_live_storage_table_client(storage_connection_str, table_name):
+    try:
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.delete_table(table_name)
+    except:
+        warnings.warn(UserWarning("storage table teardown failed"))
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    ownership_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    ownership_list = checkpoint_store.list_ownership(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(ownership_list) == 0
+
+    ownership_list = []
+
+    for i in range(ownership_cnt):
+            ownership = {}
+            ownership['fullyqualifiednamespace'] = fully_qualified_namespace
+            ownership['eventhubname'] = eventhub_name
+            ownership['consumergroup'] = consumer_group
+            ownership['ownerid'] = 'ownerid'
+            ownership['partitionid'] = str(i)
+            ownership['etag'] = None
+            ownership_list.append(ownership)
+    checkpoint_store.claim_ownership(ownership_list)
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace,eventhub_name,consumer_group)
+    assert len(ownership_list) == ownership_cnt
+
+def _update_and_list_checkpoint(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    partition_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    for i in range(partition_cnt):
+            checkpoint = {
+                'fullyqualifiednamespace': fully_qualified_namespace,
+                'eventhubname': eventhub_name,
+                'consumergroup': consumer_group,
+                'partitionid': str(i),
+                'offset': '2',
+                'sequencenumber': 20
+            }
+            checkpoint_store.update_checkpoint(checkpoint)
```

Comment:
Before you update checkpoints, can you assert that there are no checkpoints in the Table first?

Basically, we want to test that update checkpoint works both when there are no existing checkpoints, and when there are existing checkpoints, since those trigger 2 different code paths.

Reply:
okay I will do that